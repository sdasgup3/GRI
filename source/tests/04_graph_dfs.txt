define("NUM_VERTICES", "10"); // In one layer
define("NOTVISITED", "0");
define("VISITED", "1");

// Recursive depth first search
function dfs(v, dfsorder)
{
  if(v.visit == VISITED)
    return;

  println("Closing vertex: " + v.num);
  v.visit = VISITED;

  dfsorder.pushBack(v);

  foreach(neighbor; v.getNeighbors())
    dfs(neighbor, dfsorder);

}

// Generate testing graph and run DFS
function main(argv)
{
//  o 0----- o 1----- o 2----- ...o 9
//  |       |       |
//  |       |       |
//  o 10      o 11      o

  g = graph();
  v1 = g.generateVertex();
  v1.num = 0;
  first = v1;

  // Generate first layer
  for(i = 1; i < NUM_VERTICES; i++)
  {
    v2 = g.generateVertex();
    v2.num = i;

    g.generateEdge(v1, v2);
    v1 = v2;
  }

  // Generate second layer
  foreach(v1; g.getVertices())
  {
    v2 = g.generateVertex();
    v2.num = i++;
    g.generateEdge(v1, v2);
  }

  // Initialize visits
  foreach(v; g.getVertices())
    v.visit = NOTVISITED;

  // Run (any vertex first)
  dfsorder =  array(0);
  dfs(first, dfsorder);

  foreach(vertex; dfsorder) {
    println(" " + vertex.num + " ");
  }
}
