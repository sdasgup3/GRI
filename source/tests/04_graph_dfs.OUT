"Closing vertex: "0
"Closing vertex: "1
"Closing vertex: "2
"Closing vertex: "3
"Closing vertex: "4
"Closing vertex: "5
"Closing vertex: "6
"Closing vertex: "7
"Closing vertex: "8
"Closing vertex: "9
"Closing vertex: "19
"Closing vertex: "18
"Closing vertex: "17
"Closing vertex: "16
"Closing vertex: "15
"Closing vertex: "14
"Closing vertex: "13
"Closing vertex: "12
"Closing vertex: "11
"Closing vertex: "10
